src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAY_SIZE = 10;
+    private int size;
+    private T[] dataArray;
+
+    public ArrayList() {
+        this.dataArray = (T[]) new Object[DEFAULT_ARRAY_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        checkAndGrowArraySize();
+        this.dataArray[size] = value;
```

Comment:
"this" is no needed here. It will be added by default, no need to write it. Fix in other places
```suggestion
        dataArray[size] = value;
```